tests/lib/c_lib/src/main.c

```diff
@@ -598,6 +599,113 @@ void test_strtok_r(void)
 	test_strtok_r_do("1|2|3,4|5",           "| ", 5, tc01, false);
 }
 
+void test_asctime(void)
+{
+	struct tm time = {
+		.tm_sec = 0,
+		.tm_min = 1,
+		.tm_hour = 2,
+		.tm_mday = 3,
+		.tm_mon = 4,
+		.tm_year = 5,
+		.tm_wday = 6
+	};
+
+	zassert_true(strcmp(asctime(&time), "Sat May  3 02:01:00 1905\n") == 0, "asctime error");
+	time.tm_wday = -2;
+	zassert_true(strcmp(asctime(&time), "??? May  3 02:01:00 1905\n") == 0, "asctime error");
+	time.tm_mon = 15;
+	zassert_true(strcmp(asctime(&time), "??? ???  3 02:01:00 1905\n") == 0, "asctime error");
+}
+
+void test_asctime_r(void)
+{
+	struct tm time = {
+		.tm_sec = 0,
+		.tm_min = 1,
+		.tm_hour = 2,
+		.tm_mday = 3,
+		.tm_mon = 4,
+		.tm_year = 5,
+		.tm_wday = 6
+	};
+
+	char test_buf[26];
+
+	asctime_r(&time, test_buf);
+	zassert_true(strcmp(test_buf, "Sat May  3 02:01:00 1905\n") == 0, "asctime_r error");
+
+	time.tm_wday = -2;
+	asctime_r(&time, test_buf);
+	zassert_true(strcmp(test_buf, "??? May  3 02:01:00 1905\n") == 0, "asctime_r error");
+
+	time.tm_mon = 15;
+	asctime_r(&time, test_buf);
+	zassert_true(strcmp(test_buf, "??? ???  3 02:01:00 1905\n") == 0, "asctime_r error");
+}
+
+void test_div(void)
+{
+	struct input_s {
+		int num;
+		int den;
+	}
+
+	const input[] = {
+		{5, 3},
+		{5, 7},
+		{-5, 7},
+		{5, -7},
+		{-5, -7},
+	};
+
+	for (size_t i = 0; i < ARRAY_SIZE(input); ++i) {
+		int num = input[i].num;
+		int den = input[i].den;
+
+		div_t res = div(num, den);
+
+		zassert_equal(num, res.quot * den + res.rem,
+					"Invariant failed for %d / %d : q = %d, r = %d",
```

Comment:
This is still not right.  Tabs in Zephyr are eight spaces; it seems you have tabs at four spaces.  When viewed here on github you want it to look like:
```
                zassert_equal(num, res.quot * den + res.rem,
                        "Invariant failed for %ld / %ld : q = %ld, r = %ld",
                        num, den, res.quot, res.rem);
```
with 3 tabs and no spaces on the wrapped lines, or
```
                zassert_equal(num, res.quot * den + res.rem,
                              "Invariant failed for %ld / %ld : q = %ld, r = %ld",
                              num, den, res.quot, res.rem);
```
by adding spaces after the last tab to align with the first line.